wifiphisher/common/firewall.py

```diff
@@ -1,46 +1,56 @@
-#pylint: skip-file
-import subprocess
-from wifiphisher.common.constants import *
+"""Serves as an abstraction layer in front of iptables."""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from wifiphisher.common.constants import (NETWORK_GW_IP, SSL_PORT, PORT)
+from wifiphisher.common.utilities import execute_commands
 
 
 class Fw():
-    def __init__(self):
-        pass
-
-    def nat(self, internal_interface, external_interface):
-        subprocess.call(
-            ('iptables -t nat -A POSTROUTING -o %s -j MASQUERADE' %
-             (external_interface, )),
-            shell=True)
-
-        subprocess.call(
-            ('iptables -A FORWARD -i %s -o %s -j ACCEPT' %
-             (internal_interface, external_interface)),
-            shell=True)
-
-    def clear_rules(self):
-        subprocess.call('iptables -F', shell=True)
-        subprocess.call('iptables -X', shell=True)
-        subprocess.call('iptables -t nat -F', shell=True)
-        subprocess.call('iptables -t nat -X', shell=True)
-
-    def redirect_requests_localhost(self):
-        subprocess.call(
-            ('iptables -t nat -A PREROUTING -p tcp --dport 80 -j DNAT --to-destination %s:%s'
-             % (NETWORK_GW_IP, PORT)),
-            shell=True)
-        subprocess.call(
-            ('iptables -t nat -A PREROUTING -p udp --dport 53 -j DNAT --to-destination %s:%s'
-             % (NETWORK_GW_IP, 53)),
-            shell=True)
-        subprocess.call(
-            ('iptables -t nat -A PREROUTING -p tcp --dport 53 -j DNAT --to-destination %s:%s'
-             % (NETWORK_GW_IP, 53)),
-            shell=True)
-        subprocess.call(
-            ('iptables -t nat -A PREROUTING -p tcp --dport 443 -j DNAT --to-destination %s:%s'
-             % (NETWORK_GW_IP, SSL_PORT)),
-            shell=True)
+    """Handles all iptables operations."""
+
+    @staticmethod
+    def nat(internal_interface, external_interface):
+        # type: (str, str) -> None
+        """Do NAT."""
+        execute_commands([
+            "iptables -t nat -A POSTROUTING -o {} -j MASQUERADE".format(
+                external_interface),
+            "iptables -A FORWARD -i {} -o {} -j ACCEPT".format(
+                internal_interface, external_interface)
+        ])
+
+    @staticmethod
+    def clear_rules():
+        # type: () -> None
+        """Clear all rules."""
+        execute_commands([
+            "iptables -F", "iptables -X", "iptables -t nat -F",
+            "iptables -t nat -X"
+        ])
+
+    @staticmethod
+    def redirect_requests_localhost():
+        # type: () -> None
+        """Redirect all the request to local host.
```

Comment:
You can delete this line.

Reply:
due to [pep257](https://www.python.org/dev/peps/pep-0257/#id17) we need a summary line so we either have to keep the same or if you have another alternative I can change it to.

Reply:
Change it to: Redirect HTTP, HTTPS & DNS requests to localhost.